### PR TITLE
Turn off auth option

### LINKED ISSUE #90: As developer, I want an option to turn off authentication (only available when using testdata)

### PR DESCRIPTION
**Pull request by shea-maykinmedia**
Fixed #90 

Add new option to turn off authentication when using database.  Defaults to authentication being on when not using database and being off when using database but can be configured.  Will raise exception if authentication is off and not using the database.

### CODE REVIEW

File: src/openpersonen/conf/base.py [hunk 1]
```diff
@@ -411,3 +412,10 @@
 DAY_END = 8
 
 USE_STUF_BG_DATABASE = os.getenv("USE_STUF_BG_DATABASE") == "True"
+USE_AUTHENTICATION = not USE_STUF_BG_DATABASE
+
+if os.getenv("USE_AUTHENTICATION"):
+    USE_AUTHENTICATION = os.getenv("USE_AUTHENTICATION") == "True"
+
+if not USE_AUTHENTICATION and not USE_STUF_BG_DATABASE:
```
Comment (joeribekker) [inline src/openpersonen/conf/base.py:420]: This validation should not be done in settings.

File: src/openpersonen/api/tests/views/test_permissions.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            HTTP_AUTHORIZATION=f"Token {token.key}",
+        )
+        self.assertEqual(response.status_code, 200)
+
+    @override_settings(USE_STUF_BG_DATABASE=True, USE_AUTHENTICATION=False)
+    def test_not_use_authentication_and_use_database(self):
+        response = self.client.get(
+            reverse("ingeschrevenpersonen-list") + "?burgerservicenummer=123456789"
+        )
+        self.assertEqual(response.status_code, 200)
```
Comment (joeribekker) [inline src/openpersonen/api/tests/views/test_permissions.py:24]: Ok, but this should never be possible and raise an ImproperlyConfigured.

File: src/openpersonen/api/views/permissions.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+from rest_framework.permissions import IsAuthenticated as _IsAuthenticated
+
+
+class IsAuthenticated(_IsAuthenticated):
+
+    def has_permission(self, request, view):
+        if settings.USE_AUTHENTICATION:
+            return super().has_permission(request, view)
+        else:
+            return True
```
Comment (joeribekker) [inline src/openpersonen/api/views/permissions.py:11]:
```suggestion
            if not USE_STUF_BG_DATABASE:
                raise ImproperlyConfigured("Must use authentication when not using local database")
            return True
```

**Review by joeribekker — CHANGES_REQUESTED** [review]

File: src/openpersonen/api/tests/views/test_permissions.py (hunk 2 above)
Comment (shea-maykinmedia) [reply src/openpersonen/api/tests/views/test_permissions.py:24]: I think it's the last unit test that shouldn't happen and should give a 500 so I'll fix this one.  This one is testing that authentication is not needed when using the database.

**Review by joeribekker — APPROVED** [review]

**Pull request merged**